pkg/kvstore/etcd.go

```diff
@@ -520,21 +525,36 @@ func (e *etcdClient) renewSession(ctx context.Context) error {
 	// routines can get a lease ID of an already expired lease.
 	e.Lock()
 
-	timeoutCtx, cancel := context.WithTimeout(ctx, statusCheckTimeout)
-	defer cancel()
+	// Cancel any eventual old session context
+	if e.sessionCancel != nil {
+		e.sessionCancel()
+		e.sessionCancel = nil
+	}
+
+	// Create a context representing the lifetime of the session. It will
+	// timeout if the session creation does not succeed in time and then
+	// persists until:
+	//  - the parent context is cancelled due to the etcd client closing or
+	//    the controller being shut down
+	//  - The above call to sessionCancel() cancels the session due to the
```

Comment:
> It will timeout if the session creation does not succeed in time and then persists until:

Are the conditions that follow the above statement, ANDed or ORed? In other words, does the context persist when the parent context is cancelled (AND|OR) when the above call to `sessionCancel()`?